akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala

```diff
@@ -213,6 +236,9 @@ object HttpEntity {
     def withContentType(contentType: ContentType): Default =
       if (contentType == this.contentType) this else copy(contentType = contentType)
 
+    def withSizeLimit(maxBytes: Long): Default =
+      copy(data = data withAttributes Attributes(SizeLimit(maxBytes, Some(contentLength))))
```

Comment:
It always controls the wire level cap if I understand it correctly. I.e. even if you attack a Sink.ignore, which essentially would be "safe" with arbitrary large data, it will still cap, because the affected stage is down below. I think this is actually a *good* thing, but needs to be documented.